documentation/tutorials/confirmation.md

```diff
@@ -50,42 +50,86 @@ defmodule MyApp.Accounts.User do
         confirm_on_create? true
         confirm_on_update? false
         confirm_action_name :confirm_new_user
-        sender MyApp.NewUserConfirmationSender
+        sender MyApp.Accounts.User.Senders.SendNewUserConfirmationEmail
       end
     end
   end
 end
 ```
+
 Next we will have to generate and run migrations to add confirmed_at column to user resource
 
 ```bash
 mix ash.codegen account_confirmation
 ```
 
-Next we will define our "sender" module using `Swoosh`:
+To make this work we need to create a new module `MyApp.Accounts.User.Senders.SendPasswordResetEmail`:
 
 ```elixir
-defmodule MyApp.NewUserConfirmationSender do
+defmodule MyApp.Accounts.User.Senders.SendNewUserConfirmationEmail do
+  @moduledoc """
+  Sends an email confirmation email
+  """
   use AshAuthentication.Sender
+  use MyAppWeb, :verified_routes
+
+  @impl AshAuthentication.Sender
+  def send(user, token, _opts) do
+    MyApp.Accounts.Emails.deliver_email_confirmation_instructions(
+      user,
+      url(~p"/auth/user/confirm_new_user?#{[confirm: token]}")
+    )
+  end
+end
+```
+
+We also need to create a new email template:
+
+```elixir
+defmodule Example.Accounts.Emails do
+  @moduledoc """
+  Delivers emails.
+  """
+
   import Swoosh.Email
 
-  def send(user, confirm, _opts) do
-    new()
-    |> to(to_string(user.email))
-    |> from({"MyApp Admin", "support@myapp.inc"})
-    |> subject("Confirm your email address")
-    |> html_body("""
+  def deliver_email_confirmation_instructions(user, url) do
+    if !url do
+      raise "Cannot deliver confirmation instructions without a url"
+    end
+
+    deliver(user.email, "Confirm your email address", """
       <p>
-        Hi!<br />
+        Hi #{user.email},
+      </p>
 
-        Someone has tried to register a new account at <a href="https://myapp.inc">MyApp</a>.
-        If it was you, then please click the link below to confirm your identity.  If you did not initiate this request then please ignore this email.
+      <p>
+        Someone has tried to register a new account using this email address.
+        If it was you, then please click the link below to confirm your identity. If you did not initiate this request then please ignore this email.
       </p>
+
       <p>
-        <a href="https://myapp.inc/auth/user/confirm_new_user?#{URI.encode_query(confirm: confirm)}">Click here to confirm your account</a>
+        <a href="#{url}">Click here to confirm your account</a>
       </p>
     """)
-    |> MyApp.Mailer.deliver()
+  end
+
+  # For simplicity, this module simply logs messages to the terminal.
+  # You should replace it by a proper email or notification tool, such as:
+  #
+  #   * Swoosh - https://hexdocs.pm/swoosh
+  #   * Bamboo - https://hexdocs.pm/bamboo
+  #
+  defp deliver(to, subject, body) do
+    IO.puts("Sending email to #{to} with subject #{subject} and body #{body}")
```

Comment:
Do we want this `IO.puts` here?

Reply:
It didn't bother me per se. The line would probably get deleted anyway when you adapt the code to a production environment.

Reply:
true true. It probably helps with initial development if we recommend it.